OneBusAway/ui/trip_details/VehicleMapController.swift

```diff
@@ -260,4 +254,58 @@ class VehicleMapController: UIViewController, MKMapViewDelegate {
 
         return annotationView
     }
+    
+    // MARK: - UI Configurations
+    
+    func createToggleHoverBar() {
+        let blurContainer = OBAVibrantBlurContainerView.init(frame: CGRect.zero)
+        self.view.addSubview(blurContainer)
+        blurContainer.snp.makeConstraints { (make) in
+            make.width.equalTo(40)
+            make.height.equalTo(30)
+            make.right.bottom.equalToSuperview().offset(-OBATheme.defaultPadding)
+        }
+        
+        blurContainer.vibrancyEffectView.addSubview(self.toggleButton)
+        self.toggleButton.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+        self.toggleButton.addTarget(self, action: #selector(toggleButtonTapped), for: .touchUpInside)
+        self.toggleButton.isSelected = self.expanded
+        
+    }
+    
+    
+
+    func createLocationHoverBar() {
+        let recenterMapButton = UIBarButtonItem(image: UIImage(named: "Map_Selected"), landscapeImagePhone: nil, style: .plain, target: self, action: #selector(self.recenterMap))
+        
+        //instantiate and customize location hover bar
+        let locationHoverBar = ISHHoverBar()
+        locationHoverBar.backgroundColor = UIColor.white
+        locationHoverBar.alpha = 0.7;
+        locationHoverBar.borderColor = UIColor.gray
+        locationHoverBar.borderWidth = 1.0
+        locationHoverBar.cornerRadius = 4.3
+        locationHoverBar.items = [recenterMapButton]
+        view.addSubview(locationHoverBar)
+       
+        //UI: zoom to location button appears in toggle expanded view
+        //UX: selected bus departure time cell is transparent for a spacious Map View
+        locationHoverBar.snp.makeConstraints { (make) in
+            //set hugging frame constraints
+            make.edges.equalToSuperview().inset(UIEdgeInsetsMake(263, 365.4, 50, 8))
+            //transparent selected bus cell
+            make.bottom.equalTo(mapView).inset(UIEdgeInsetsMake(0, 0, 99.8, 0))
+        }
+   
+    }
+    
+    //runs on an asynchronous GCD user initiated priority queue
+    func recenterMap() {
+        DispatchQueue.global(qos: DispatchQoS.userInitiated.qosClass).async {
```

Comment:
Calling UI components on a background thread or queue will cause crashes. This shouldn't need to be run within an async operation anyway.